Can't only migrate remaining versions

The script died on me 1/2 way through migration and then I couldn't run it a 2nd time to only migrate the missing versions.

Would be good to change this so it fails gracefully on some versions already being migrated instead of throwing this error:

```js
npm ERR! publish Failed PUT 403
{ Error: Forbidden The feed already contains the package 'my-module' at version '0.1.2': _packaging
    at makeError (/Users/bblackwo/dev/node/npm-migrate/node_modules/npm/node_modules/npm-registry-client/lib/request.js:304:12)
    at CachingRegistryClient.<anonymous> (/Users/bblackwo/dev/node/npm-migrate/node_modules/npm/node_modules/npm-registry-client/lib/request.js:292:14)
    at Request._callback (/Users/bblackwo/dev/node/npm-migrate/node_modules/npm/node_modules/npm-registry-client/lib/request.js:212:14)
    at Request.self.callback (/Users/bblackwo/dev/node/npm-migrate/node_modules/npm/node_modules/request/request.js:186:22)
    at emitTwo (events.js:126:13)
    at Request.emit (events.js:214:7)
    at Request.<anonymous> (/Users/bblackwo/dev/node/npm-migrate/node_modules/npm/node_modules/request/request.js:1081:10)
    at emitOne (events.js:116:13)
    at Request.emit (events.js:211:7)
    at IncomingMessage.<anonymous> (/Users/bblackwo/dev/node/npm-migrate/node_modules/npm/node_modules/request/request.js:1001:12) pkgid: '_packaging', statusCode: 403, code: 'E403' }
```

--- a/.editorconfig
+++ b/.editorconfig
@@ -1,0 +1,16 @@
+# http://EditorConfig.org
+
+# top-most EditorConfig file
+root = true
+
+# Unix-style newlines with a newline ending every file
+[*]
+end_of_line = lf
+insert_final_newline = true
+
+# Set default charset
+charset = utf-8
+
+# Indentation
+indent_style = space
+indent_size = 4

--- a/README.md
+++ b/README.md
@@ -26,7 +26,7 @@ migrate(moduleName, from, to, options)
 
 ## What it does
 
-1. Fetches all versions as tarballs from old registry
+1. Fetches all versions (or just those not already migrated) as tarballs from old registry
 2. Extracts & updates `package.json`: the `publishConfig.registry` field to the new registry url
 3. Publishes each version to the new registry
 4. Cleans up after itself
@@ -35,3 +35,8 @@ migrate(moduleName, from, to, options)
 
 - The dates of every version published will be reset to the date and time you run this script
 - The migrating user will be added as maintainer
+
+## Changelog
+
+- v1.2.0 - Work with scoped packages
+- v1.3.0 - Compare both registries and migrate only the remaining versions not in the new registry

--- a/npm-migrate.js
+++ b/npm-migrate.js
@@ -22,7 +22,7 @@ module.exports = function (moduleName, oldRegistry, newRegistry, options = { deb
     let curried_publishSeries = curry(publishSeries)
     curried_publishSeries = curried_publishSeries(newRegistry)
 
-    return getVersionList(moduleName, oldRegistry)
+    return getVersionList(moduleName, oldRegistry, newRegistry)
         .then(curried_getTarballs)
         .then(unpack)
         .then(curried_updatePackage)

--- a/npm_utils.js
+++ b/npm_utils.js
@@ -51,20 +51,57 @@ const getTarball = function (moduleName, registry, version, callback) {
 
 }
 
-module.exports.getVersionList = function (moduleName, registry) {
+function getRemainingVersions (moduleName, oldRegistry, newRegistry, oldRegistryVersions) {
+
+    return new Promise((resolve) => {
+
+        npm.config.set('registry', newRegistry);
+        npm.commands.info([moduleName], (err, data) => {
+
+            if (err) {
+                return resolve(oldRegistryVersions);
+            }
+        
+            const latest = Object.keys(data)[0];
+            const newRegistryVersions = data[latest].versions;
+            console.log('New Registry Versions', newRegistryVersions);
+
+            remainingVersions = oldRegistryVersions.filter((v) => !newRegistryVersions.includes(v));
+            if (!remainingVersions.length) {
+                console.log('No more versions of this package to migrate');
+            } else {
+                console.log('Remaining Versions to Migrate', remainingVersions);
+            }
+
+            resolve(remainingVersions);
+
+        });
+        npm.config.set('registry', oldRegistry);
+
+  });
+
+}
+
+module.exports.getVersionList = function (moduleName, oldRegistry, newRegistry) {
 
     return new Promise((resolve, reject) => {
 
         npm.load({
-            registry: registry
+            registry: oldRegistry
         }, () => {
 
             npm.commands.info([moduleName], function (err, data) {
 
                 if (err) return reject(err);
 
-                let latest = Object.keys(data)[0];
-                resolve(data[latest].versions)
+                const latest = Object.keys(data)[0];
+                const oldRegistryVersions = data[latest].versions;
+                console.log('Old Registry Versions', oldRegistryVersions);
+
+                return getRemainingVersions(moduleName, oldRegistry, newRegistry, oldRegistryVersions)
+                    .then((remainingVersions) => {
+                        resolve(remainingVersions);
+                    });
             })
         })
     })

--- a/package.json
+++ b/package.json
@@ -1,6 +1,6 @@
 {
   "name": "npm-migrate",
-  "version": "1.2.0",
+  "version": "1.3.0",
   "description": "Migrate all versions of a selected package from a registry to another one",
   "main": "npm-migrate.js",
   "scripts": {

